include/mbedtls/sha3.h

```diff
@@ -0,0 +1,262 @@
+ *                 This must be a writable buffer of length \c olen bytes.
+ * \param olen     Defines a variable output length (in bytes). \c output must be
+ *                 \c olen bytes length. For SHA-3 224, SHA-3 256, SHA-3 384 and
+ *                 SHA-3 512 must equal to 28, 32, 48 and 64, respectively.
+ *                 For SHAKE128 and SHAKE256 it can be an arbitrary number.
+ *
+ * \return         \c 0 on success.
+ * \return         A negative error code on failure.
+ */
+int mbedtls_sha3_finish( mbedtls_sha3_context *ctx,
+                           uint8_t *output, size_t olen );
+
+/**
+ * \brief          This function calculates the SHA-3
+ *                 checksum of a buffer.
+ *
+ *                 The function allocates the context, performs the
+ *                 calculation, and frees the context.
+ *
+ *                 The SHA-3 result is calculated as
+ *                 output = SHA-3(id, input buffer, d).
+ *
+ * \param id       The id of the SHA-3 family.
+ * \param input    The buffer holding the data. This must be a readable
+ *                 buffer of length \p ilen Bytes.
+ * \param ilen     The length of the input data in Bytes.
+ * \param output   The SHA-3 checksum result.
+ *                 This must be a writable buffer of length \c olen bytes.
+ * \param olen     Determines the length (in bytes) of the output. \c output
+ *                 must be \c olen bytes length.
+ *
+ * \return         \c 0 on success.
+ * \return         A negative error code on failure.
+ */
+int mbedtls_sha3( mbedtls_sha3_id id, const uint8_t *input,
+                    size_t ilen,
+                    uint8_t *output,
+                    size_t olen );
+
+/**
+ * \brief          This function calculates the SHA-3
+ *                 checksum of a buffer with function-name and customization
+ *                 strings.
+ *
+ *                 The function allocates the context, performs the
+ *                 calculation, and frees the context.
+ *
+ *                 The SHA-3 result is calculated as
+ *                 output = SHA-3(id, name || customization || input buffer, d).
+ *
+ * \param id       The id of the SHA-3 family. It can be \c MBEDTLS_SHA3_CSHAKE128 or
+ *                 \c MBEDTLS_SHA3_CSHAKE256.
+ * \param input    The buffer holding the data. This must be a readable
+ *                 buffer of length \p ilen Bytes.
+ * \param ilen     The length of the input data in Bytes.
+ * \param name     The function-name string. It must be zero-terminated.
+ * \param name_len The length of \c name in Bytes.
+ * \param custom   The customization string. It must be zero-terminated.
+ * \param custom_len The length of \c custom in Bytes.
+ * \param output   The SHA-3 checksum result.
+ *                 This must be a writable buffer of length \c olen bytes.
+ * \param olen     Determines the length (in bytes) of the output. \c output
+ *                 must be \c olen bytes length.
```

Comment:
The olen argument isn't as adequately described as in the update function. Perhaps once we determine the correct text for that, it can be used here as well.